runtime/lib/ttnn/operations/kv_cache/update_cache.cpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "update_cache.h"
+#include "tt/runtime/detail/logger.h"
+#include <ttnn/operations/core/core.hpp>
+#include <ttnn/operations/data_movement/slice/slice.hpp>
+
+namespace tt::runtime::ttnn::operations::kv_cache {
+void run(const ::tt::target::ttnn::UpdateCacheOp *op, ProgramContext &context) {
+
+  ProgramTensorPool &tensorPool = context.getTensorPool();
+
+  const ::ttnn::Tensor &cache = tensorPool.at(op->cache()->global_id());
+  const ::ttnn::Tensor &input = tensorPool.at(op->input()->global_id());
+  const ::ttnn::Tensor &updateIndex =
+      tensorPool.at(op->update_index()->global_id());
+
+  const ::ttnn::Tensor index_on_host = ::ttnn::from_device(updateIndex);
+  const auto storage = index_on_host.get_storage();
+  const auto ownedStorage = std::get<tt_metal::OwnedStorage>(storage);
+  const auto buffer = ownedStorage.get_buffer();
+  const auto buf = std::get<tt_metal::owned_buffer::Buffer<uint32_t>>(buffer);
+  uint32_t up_idx = *buf.begin();
```

Comment:
We should probably find a better way to do this, or maybe mark it as a workaround